examples/tests/monte-carlo.cpp

```diff
@@ -0,0 +1,88 @@
+#include "gtest/gtest.h"
+
+#include <gstorm.h>
+#include <vector>
+#include <cmath>
+#include <functional>
+#include <range/v3/all.hpp>
+
+#include "experimental.h"
+
+struct MonteCarlo : public testing::Test {};
+
+#ifdef __SYCL_DEVICE_ONLY__
+extern "C" {
+float logf(float f) {
+  return cl::sycl::log(f);
+}
+
+long double logl(long double f) {
+  return cl::sycl::log(cl_double(f));
+}
+
+float nextafterf(float f, float g) {
+  return cl::sycl::nextafter(f, g);
+}
+}
+#endif
+
+struct estimate_pi {
+  float operator()(unsigned int thread_id) const {
+    float sum = 0;
+    unsigned int N = 5000; // samples per stream
+
+    // Workaround
+    nextafterf(1.f,2.f);
+    logl(1.f);
+    sqrtf(1.f);
+    logf(1.f);
+
+    // create a random number generator
+    std::default_random_engine rng(thread_id);
+
+    // create a mapping from random numbers to [0,1)
+    std::uniform_real_distribution<float> u01(0, 1);
```

Comment:
is this working on the GPU device? I am surprised

Reply:
Not right out of the box, but yes. @DuncanMcBain came up with the tweaks to make it work.